core/src/main/scala/org/apache/spark/memory/MemoryManager.scala

```diff
@@ -267,4 +281,22 @@ private[spark] abstract class MemoryManager(
       case MemoryMode.OFF_HEAP => MemoryAllocator.UNSAFE
     }
   }
+
+  /**
+   * Return whether we are using G1GC or not
+   */
+  private[memory] final val isG1GC: Boolean = {
```

Comment:
```suggestion
  private final val isG1GC: Boolean = {
```

Reply:
Should we make it `lazy` too to fully leverage `private lazy val defaultPageSizeBytes` you wrote?

Reply:
fine, updated